Sources/Plasma/CoreLib/hsThread_Unix.cpp

```diff
@@ -106,6 +106,8 @@ hsGlobalSemaphore::hsGlobalSemaphore(int initialValue, const ST::string& name)
 
         /* Named semaphore shared between processes */
         fPSema = sem_open(semName.c_str(), O_CREAT, 0666, initialValue);
+        // Unlink it immediately so it will be freed if we unexpectedly leave it locked
+        sem_unlink(semName.c_str());
```

Comment:
I don't think this is correct. This will cause other processes to get a new semaphore that isn't connected to this semaphore, breaking the process shared nature of the global semaphore.

Reply:
That may actually be the desired behavior here. The Windows version, when used with logging, creates a process local semaphore. Not a machine local semaphore.

However:
- That may not have been what was intended on Windows.
- The Windows version can still create a global semaphore if the semaphore name is prepended with 'global\' (have not tested- that's what the docs claim.)

Certainly open to suggestions here. Not seeing a great way to close the semaphore automatically on an unexpected exit.

Reply:
Yeah, I have mixed thoughts on this right now... on one hand, this is the easy solution that solves the common case, and I don't think we currently have any cases where we need multi-process semaphores, but this does potentially introduce a limitation where semaphores can't actually be used across multiple processes

Reply:
Empirical evidence suggests that Win32 semaphores are created in either the global or session namespace without the local namespace prefix. That is basically the point of the `hsGlobalSemaphore` - to be able to wait and signal across processes. The big difference is that Win32 semaphores automatically destroy themselves when the last process holding a `HANDLE` to the semaphore terminates. I'm not really sure what to do here, unfortunately. We may need to rethink how we synchronize access to log files when multiple clients are running.

Reply:
Yeah. Just to be clear here - not a shot in the dark. When @dpogue was describing the didn't-launch-til-restart behavior I knew exactly what it was because I've seen it happen before.

It shouldn't happen in normal operation but every once in a while if the client crashes it just gets stuck. Especially worrisome - if the client locked up due to a deadlock it will stay deadlocked when it relaunches.

Reply:
I don't doubt that this fixes the deadlock on launch at all. The problem is that this fixes it by break in the global semaphore mechanism.